desktop-exporter/app/routes/trace-view.jsx

```diff
@@ -40,22 +44,29 @@ function WaterfallView(props) {
     );
 }
 
-function DetailView() {
+function DetailView(props) {
     return (
         <div className='detail'>
-            Details will go here
+            Name: {props.spanData.name} <br />
+            Kind: {props.spanData.kind} <br />
+            Start: {props.spanData.startTime} <br />
+            End: {props.spanData.endTime} <br />
         </div>
-    )
+    );
 }
 
 export default function TraceView() {
     const traceData = useLoaderData();
+    const [activeIndex, setActiveIndex] = React.useState(0);
+    const handleOnClick = index => {
+        setActiveIndex(index);
+    };
```

Comment:
You don't need this function indirection :) `handleOnClick` here does exactly what `setActiveIndex` does.

The code below does the same.

```js
<WaterfallView spans={traceData.spans} clickHandler={setActiveIndex} activeIndex={activeIndex} />
```

Reply:
Oh cool!